### PR TITLE
Allow changing text language from the properties palette

### PR DESCRIPTION
**Pull request by asmaAL-Bahanta**
Allow changing text language from the properties palette
and show the whole language list in style manager.

### CODE REVIEW

File: scribus/plugins/scriptplugin/cmdtext.h [hunk 1]
```diff
@@ -197,6 +197,19 @@ May throw ValueError if the font cannot be found.\n\
 /*! Set font features */
 PyObject *scribus_setfontfeatures(PyObject * /*self*/, PyObject* args);
 
+ /*! docstring */
+PyDoc_STRVAR(scribus_setlanguage__doc__,
+QT_TR_NOOP("setlanguage(\"language\", [\"name\"])\n\
+\n\
+Sets the font of the text frame \"name\" to \"language\". If there is some text\n\
```
Comment (khaledhosny) [inline scribus/plugins/scriptplugin/cmdtext.h]: This should be `set the language of…`, but why do we need any changes to python here?

File: scribus/text/textshaper.cpp [hunk 2]
```diff
@@ -270,7 +270,6 @@ QList<GlyphRun> TextShaper::shape()
 		hb_buffer_set_direction(hbBuffer, hbDirection);
 		hb_buffer_set_script(hbBuffer, hbScript);
 		hb_buffer_set_language(hbBuffer, hbLanguage);
-		hb_buffer_set_cluster_level(hbBuffer, HB_BUFFER_CLUSTER_LEVEL_MONOTONE_CHARACTERS);
```
Comment (khaledhosny) [inline scribus/text/textshaper.cpp]: Please return this accidentally removed line.

File: scribus/styles/paragraphstyle.h [hunk 3]
```diff
@@ -148,14 +148,12 @@ class SCRIBUS_API ParagraphStyle : public Style
...
 #undef ATTRDEF
-	
-	
+	CharStyle m_cstyle;
 private:
 		
 	// member declarations:
 	StyleContextProxy m_cstyleContext;
 	bool m_cstyleContextIsInh;
-	CharStyle m_cstyle;
```
Comment (khaledhosny) [inline scribus/styles/paragraphstyle.h]: Why was this private member made public? If you really need it add an accessor function.

File: scribus/ui/prefs_hyphenator.cpp [hunk 4]
```diff
@@ -33,7 +33,7 @@ Prefs_Hyphenator::Prefs_Hyphenator(QWidget* parent, ScribusDoc* doc)
 	setupUi(this);
 
 	QStringList languageList;
-	LanguageManager::instance()->fillInstalledHyphStringList(&languageList);
```
Comment (khaledhosny) [inline scribus/ui/prefs_hyphenator.cpp]: Why are you changing the hyphenation preferences? We shouldn’t need any changes here.

File: scribus/ui/smcstylewidget.cpp [hunk 5]
```diff
@@ -322,27 +322,26 @@ void SMCStyleWidget::show(CharStyle *cstyle, QList<CharStyle> &cstyles, const QS
 	}
 
 	QString clang = cstyle->language().isNull() || cstyle->language().isEmpty() ?
-	                                      defLang : cstyle->language();
```
Comment (khaledhosny) [inline scribus/ui/smcstylewidget.cpp]: Please remove the unnecessary formating change to this line.

File: scribus/ui/smcstylewidget.cpp [hunk 6]
```diff
@@ -322,27 +322,26 @@ void SMCStyleWidget::show(CharStyle *cstyle, QList<CharStyle> &cstyles, const QS
...
+                                             defLang : cstyle->language();
 	//qDebug()<<"style lang"<<cstyle->language()<<clang;
 	QString plang(QString::null);
 	if (hasParent)
 		plang = parent->language().isNull() || parent->language().isEmpty() ?
-		                              defLang : parent->language();
```
Comment (khaledhosny) [inline scribus/ui/smcstylewidget.cpp]: Same as above.

**Comment by asmaAL-Bahanta** [comment]
Request updated.

**Pull request merged**